Sources/Apollo/ApolloStore.swift

```diff
@@ -206,74 +208,86 @@ public final class ApolloStore {
     fileprivate let cache: NormalizedCache
     fileprivate let cacheKeyForObject: CacheKeyForObject?
 
-    fileprivate lazy var loader: DataLoader<CacheKey, Record?> = DataLoader(self.cache.loadRecordsPromise)
+    fileprivate lazy var loader: DataLoader<CacheKey, RecordRow?> = DataLoader(self.cache.loadRecordsPromise)
 
     init(cache: NormalizedCache, cacheKeyForObject: CacheKeyForObject?) {
       self.cache = cache
       self.cacheKeyForObject = cacheKeyForObject
     }
 
-    public func read<Query: GraphQLQuery>(query: Query) throws -> Query.Data {
+    public func read<Query: GraphQLQuery>(query: Query) throws -> (Query.Data, GraphQLResultContext) {
```

Comment:
I feel returning a tuple makes the use of the store read API a bit awkward, because you're often just interested in the data, and now you have to destructure the returned tuple explicitly everywhere. Not sure if the alternative is any better though. I was thinking of returning `GraphQLResult` from all store read methods instead, but that means you have to go through `data` even if you're doing a `readObject` for an individual fragment.

Reply:
Maybe a better question to ask is: do we need the public read store API to provide access to metadata at all? You can always use a `fetch` with `.returnCacheDataDontFetch` to read a result from the cache. I think the main purpose of the store read API is for manual store manipulation within writes, so that may not require the metadata?